react/ThumbnailItem.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+class ThumbnailItem extends Component {
+  handleClick(event) {
+    event.preventDefault()
+
+    const { image, onThumbnailClick } = this.props
+
+    if (onThumbnailClick) {
+      onThumbnailClick(image)
+    }
+  }
+
+  render() {
+    const { image, onThumbnailClick } = this.props
+    const { imageUrl, imageText } = image
+    
+    return (
+      <div className="dtr w3">
+        <a href="#" className="dim">
+          <img src={imageUrl} alt={imageText} onClick={this.handleClick.bind(this)} />
```

Comment:
Try to bind using the arrow notation.
https://medium.com/@machnicki/handle-events-in-react-with-arrow-functions-ede88184bbb